FirebaseAuth/Sources/Swift/SystemService/AuthNotificationManager.swift

```diff
@@ -140,12 +136,12 @@
         return false
       }
       if dictionary[kNotificationProberKey] != nil {
-        if pendingCallbacks == nil {
+        if hasCheckedNotificationForwarding {
           // The prober notification probably comes from another instance, so pass it along.
           return false
         }
         isNotificationBeingForwarded = true
```

Comment:
I think I see one potential data race. Based on how the order of public APIs called for this flow, I don't think it'd likely. 

This method is called only by:
<img width="621" alt="Screenshot 2024-12-10 at 10 53 08 AM" src="https://github.com/user-attachments/assets/008ccf19-08eb-44f7-932b-1475a9dd97b6">

So it's always run from the `kAuthGlobalWorkQueue` context.

`checkNotificationForwarding` is being called non on the `kAuthGlobalWorkQueue`
<img width="1027" alt="Screenshot 2024-12-10 at 12 19 42 PM" src="https://github.com/user-attachments/assets/da8b7f5e-4fe8-4987-af25-00044d673f98">

Both methods depend on the `hasCheckedNotificationForwarding` and `isNotificationBeingForwarded` so multiple calls to `checkNotificationForwarding` could lead to a data race since the boolean properties are not atomic.

Reply:
Thanks - since this is unlikely to occur given how the methods are called and the PR is an incremental improvement, resolving and merging.